External/Plugins/FlashDebugger/Helpers/CopyAsTree.cs

```diff
@@ -0,0 +1,212 @@
+
+				// recurse for children .. but skip if unwanted items
+				if (child.Nodes.Count > 0 && IsWantedParent(child))
+				{
+
+					// opening brace
+					sb.AppendLine("{");
+
+					string childID = child.ID;
+
+					// add error if too many levels of recursion
+					if (level <= levelLimit || levelLimit == 0)
+					{
+
+						// check if encountered before
+						bool isNew = !CopiedObjects.ContainsKey(childID);
+						if (!isNew)
+						{
+
+							// error
+							AppendTimes(sb, levelSep, level + 1);
+							sb.AppendLine("[Already listed before]");
+
+						}
+						else if (level > CopyTreeMaxRecursion)
+						{
+
+							// error
+							AppendTimes(sb, levelSep, level + 1);
+							sb.AppendLine("[Recursion too deep, increase CopyTreeMaxRecursion in FlashDebugger settings]");
+
+						}
+						else
+						{
+
+							// add to list
+							CopiedObjects.Add(childID, 1);
+
+							// children
+							try
+							{
+								GetTreeItemsAsText(childTree.Children, child.Nodes, levelSep, level + 1, sb, control, levelLimit);
+							}
+							catch (System.Exception ex) { }
+
+						}
+					}
+
+
+					// closing brace
+					AppendTimes(sb, levelSep, level);
+					sb.AppendLine("}");
+
+				}
+				else
+				{
+					sb.AppendLine();
+				}
+
+
+				// stop recursion if too long
+				if (sb.Length > CopyTreeMaxChars)
+				{
+					sb.Append("......");
+					return;
+				}
+
+
+			}
+		}
+
+		private static bool IsWantedParent(DataNode parent)
+		{
+			try
+			{
+
+				// if is an empty array []
+				// then skip it from opening and closing { } the output
+				if (parent.Nodes.Count == 1 || parent.Nodes.Count == 2)
+				{
+					if (parent.Value.StartsWith("Array"))
+					{
+						DataNode child1 = (DataNode)parent.Nodes[0];
+						if (child1.Text == "[static]" || child1.Text == "length")
+						{
+							return false;
+						}
+					}
+				}
+
+				// catch "cannot cast Aga.TreeNode to DataNode" 
+				// TODO : fix this instead of just catching it
+			}
+			catch (Exception ex) { }
+			return true;
+		}
+
+		private static string[] as3DisabledProps = new string[] { "stage", "parent", "root", "loaderInfo", "_nativeWindow", "nativeWindow" };
+
+		private static bool IsWantedChild(DataNode child)
+		{
+			try
+			{
+				if (child.Parent != null)
+				{
+					DataNode parent = ((DataNode)child.Parent);
+
+					// if is an array []
+					// skip [static] and "length" properties
+					if (parent.Value.StartsWith("Array"))
+					{
+						if (child.Text == "[static]" || child.Text == "length")
+						{
+							return false;
+						}
+					}
+
+					// if is an AS3 display object,
+					// don't go upward (stage, parent)
+					if (parent.Value.StartsWith("flash.display.") || parent.Value == "Main")
+					{
+						if (Array.IndexOf(as3DisabledProps, child.Text) > -1)
+						{
+							return false;
+						}
+					}
+
+					// if is an AS2 display object,
+					// don't go upward (_parent)
+					if (parent.Value == "MovieClip" || parent.Value == "Video")
```

Comment:
As before, the user could have a custom class named MovieClip, Video, Main, define classes inside flash.display, etc and have wanted variables named like the ones being filtered out.

As I said, chances are slim, but they are there.